File: 2265-partition-array-according-to-given-pivot/2265-partition-array-according-to-given-pivot.java
class Solution {
    public int[] pivotArray(int[] nums, int pivot) {
        int count=0;
        int index=0;
        int[] ans=new int[nums.length];

        for(int num:nums){
            if(num==pivot) count++;
            else if(num<pivot) {
                ans[index]=num;
                index++;
            }
        }

        while(count>0){
            ans[index]=pivot;
            index++;
            count--;
        }

        for(int num:nums){
            if(num>pivot){
                ans[index]=num;
                index++;
            }
        }

        return ans;
    }
}

File: 2265-partition-array-according-to-given-pivot/2265-partition-array-according-to-given-pivot.cpp
class Solution {
public:
    vector<int> pivotArray(vector<int>& nums, int pivot) {
        int count=0;
        vector<int>ans;
        for(int i=0;i<nums.size();i++){
            if(nums[i]==pivot) count++;
            else if(nums[i]<pivot) ans.push_back(nums[i]);
        }
        while(count!=0){
            ans.push_back(pivot);
            count--;
        }
        for(int i=0;i<nums.size();i++){
            if(nums[i]>pivot){
                ans.push_back(nums[i]);
            }
        }
        return ans;
    }
};